ang/prospect/case-actions/services/unlink-contribution-case-action.service.js

```diff
@@ -0,0 +1,52 @@
+(function (angular) {
+  var module = angular.module('prospect');
+
+  module.service('UnlinkContributionCaseAction', UnlinkContributionCaseAction);
+
+  /**
+   * @param {object} ProspectConverted Prospect Converted Service
+   */
+  function UnlinkContributionCaseAction (ProspectConverted) {
+    /**
+     * Checks if the Action is allowed
+     *
+     * @param {object} action action
+     * @param {Array} cases cases
+     * @returns {boolean} if action is allowed
+     */
+    this.isActionAllowed = function (action, cases) {
+      if (!cases[0] || !cases[0].prospect.paymentInfo) {
+        return;
```

Comment:
This maybe is my lack of javascript knowledge, but let me ask: this is exactly the same as putting `return false`? I would prefer the latter.

Reply:
Yes, `return` or `return false` or `return undefined`, all are same in Javascript. And because Javascript is not a typed language, it is a common practise to just write `return`. Also the same approach is heavily used in all CPS products.
So while I agree with you, that `return false` is better. But on the other hand, its an opinionated view, which to be honest most JS developers do not follow.
And we have a style guide called `semistandard` that we use for JS linting, and it does not enforce https://eslint.org/docs/rules/consistent-return rule. Hence I would say lets leave this to the choice of the developer who is writing the code. If someone chooses to mention consistent return type, all good, and if they dont, thats also good.

TLDR: This is an opinionated view, and currently the JS linter used in this project does not enforce this.

Reply:
thanks for the explanation! I understand now